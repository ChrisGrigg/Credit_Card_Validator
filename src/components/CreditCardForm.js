import React from 'react';
import {func} from 'prop-types';
import CreditCardResults from './CreditCardResults';
import CreditCardTextInput from './CreditCardTextInput';
import {cardValidation} from '../types';

const CreditCardForm = ({cardValidation, onChange}) => (
  <div>
    <h2>Credit Card validator</h2>
    <table>
      <tbody>
        <tr>
          <td><label htmlFor="cardNumber">Credit Card No.</label></td>
          <td><CreditCardTextInput onChange={onChange} name="cardNumber" value={cardValidation.cardNumber}/>
          </td>
        </tr>
      </tbody>
    </table>

    <hr/>

    {cardValidation.requiredDataProvidedForValidation && <CreditCardResults cardNumberValid={cardValidation.cardNumberValid} />}
  </div>
);

CreditCardForm.propTypes = {
  onChange: func.isRequired,
  cardValidation: cardValidation.isRequired
};

export default CreditCardForm;
