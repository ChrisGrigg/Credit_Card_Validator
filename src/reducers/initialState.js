export default {
  cardValidation: {
    cardNumber: '',
    displayResults: false,
    requiredDataProvidedForValidation: false,
    cardNumberValid: {
      cardNumber: '',
      cardType: '',
      valid: false
    }
  }
};
