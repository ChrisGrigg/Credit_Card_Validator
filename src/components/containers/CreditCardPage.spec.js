import React from "react";
import { mount, shallow } from "enzyme";
import configureMockStore from "redux-mock-store";
import { Provider } from "react-redux";
import { create } from "react-test-renderer";
import ConnectedCreditCardPage, { CreditCardPage } from "./CreditCardPage";
import CreditCardForm from "../CreditCardForm";
import initialState from "../../reducers/initialState";

describe("<CreditCardPage />", () => {
  const actions = {
    validateCreditCard: jest.fn()
  };

  it("should contain <CreditCardForm />", () => {
    const wrapper = shallow(
      <CreditCardPage
        actions={actions}
        cardValidation={initialState.cardValidation}
      />
    );

    expect(wrapper.find(CreditCardForm).length).toEqual(1);
  });

  it("calls validateCreditCard upon changing a field", () => {
    const wrapper = mount(
      <CreditCardPage
        actions={actions}
        cardValidation={initialState.cardValidation}
      />
    );
    const name = "cardNumber";
    const value = '4111111111111111';

    const input = wrapper.find('input[name="cardNumber"]');
    input.simulate("change", { target: { name, value } });

    expect(actions.validateCreditCard).toHaveBeenCalledWith(
      initialState.cardValidation,
      name,
      value
    );
  });

  it("should match snapshot", () => {
    const store = configureMockStore()(initialState);
    const component = create(
      <Provider store={store}>
        <ConnectedCreditCardPage />
      </Provider>
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
