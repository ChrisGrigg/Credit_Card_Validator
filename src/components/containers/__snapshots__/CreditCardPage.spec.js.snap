// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreditCardPage /> should match snapshot 1`] = `
<div>
  <h2>
    Credit Card validator
  </h2>
  <table>
    <tbody>
      <tr>
        <td>
          <label
            htmlFor="cardNumber"
          >
            Credit Card No.
          </label>
        </td>
        <td>
          <input
            className="medium"
            name="cardNumber"
            onChange={[Function]}
            placeholder={undefined}
            type="text"
            value=""
          />
        </td>
      </tr>
    </tbody>
  </table>
  <hr />
</div>
`;
